packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts

```diff
@@ -42,45 +42,56 @@ async function main() {
   const grid3 = await getClient(`kubernetes/${name}`);
 
   const qsfs_name = "testQsfsK8sq1";
+  const disk_size = 1;
+  const count = 8;
 
-  const masterQueryOptions: FilterOptions = {
+  const options: FilterOptions = {
     cru: 2,
     mru: 2, // GB
     sru: 6,
     availableFor: grid3.twinId,
     farmId: 1,
   };
 
-  const workerQueryOptions: FilterOptions = {
-    cru: 1,
-    mru: 1, // GB
-    sru: 3,
-    availableFor: grid3.twinId,
-    farmId: 1,
-  };
-
   const qsfsQueryOptions: FilterOptions = {
     hru: 15,
     availableFor: grid3.twinId,
     farmId: 1,
   };
 
   const qsfsNodes: number[] = [];
+  let usedResources = 0;
+
+  const nodes = await grid3.capacity.filterNodes(qsfsQueryOptions);
+
+  // Loop over filtered nodes and push in qsfsNodes until they have the required resources
+  for (const node of nodes) {
+    const freeResources = await grid3.capacity.nodes.getNodeFreeResources(node.nodeId);
+    const freeHRU = freeResources.hru / 1024 ** 3;
+
+    if (freeHRU >= disk_size && usedResources <= disk_size * count) {
+      qsfsNodes.push(node.nodeId);
+      usedResources += freeHRU;
+    }
+
+    if (usedResources >= disk_size * count) {
+      break;
+    }
+  }
```

Comment:
why do we need that? the filter nodes method is already getting nodes with requested resources